tests/virtual_machines_test.go

```diff
@@ -41,44 +36,428 @@ var _ = Describe("Virtual Machines", func() {
+
+				vm2 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("ovs2", "")}, []kubevirtv1.Network{newNetwork("ovs2")})
+
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), vm2)
+
+				}, 40*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+				Expect(vm2.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress).ToNot(BeEmpty())
+
+				_, err = net.ParseMAC(vm2.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+
+				vm1MacAddress := vm1.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+				vm2MacAddress := vm2.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress
+
+				//deleting both and try to assign their MAC address to the new VM
+				err = testClient.VirtClient.Delete(context.TODO(), vm1)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = testClient.VirtClient.Delete(context.TODO(), vm2)
+				Expect(err).ToNot(HaveOccurred())
+
+				newVM1 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("newOvs1", vm1MacAddress)},
+					[]kubevirtv1.Network{newNetwork("newOvs1")})
+
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), newVM1)
+
+				}, 40*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+
+				_, err = net.ParseMAC(newVM1.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+
+				newVM2 := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("newOvs2", vm2MacAddress)},
+					[]kubevirtv1.Network{newNetwork("newOvs2")})
+
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), newVM2)
+
+				}, 40*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+
+				_, err = net.ParseMAC(newVM2.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+		//2162
+		Context("When trying to create a VM after all MAC addresses in range have been occupied", func() {
+			It("should return an error because no MAC address is available", func() {
+
+				err := setRange("02:00:00:00:00:00", "02:00:00:00:00:01")
+				Expect(err).ToNot(HaveOccurred())
+
+				vm := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("ovs1", ""),
+					newInterface("ovs2", "")}, []kubevirtv1.Network{newNetwork("ovs1"), newNetwork("ovs2")})
+
+				Eventually(func() error {
+					return testClient.VirtClient.Create(context.TODO(), vm)
+
+				}, 40*time.Second, 5*time.Second).Should(Not(HaveOccurred()), "failed to apply the new vm object")
+
+				_, err = net.ParseMAC(vm.Spec.Template.Spec.Domain.Devices.Interfaces[0].MacAddress)
+				Expect(err).ToNot(HaveOccurred())
+
+				starvingVM := CreateVmObject(TestNamespace, false, []kubevirtv1.Interface{newInterface("starvingOvs", "")},
```

Comment:
no need for the interface name to be `starvingOvs` it can be anything so maybe `ovs` should be fine